src/QueryPipeline/Pipe.cpp

```diff
@@ -817,15 +817,18 @@ void Pipe::transform(const Transformer & transformer)
         throw Exception("Transformation of Pipe is not valid because processors don't have any "
                         "not-connected output ports", ErrorCodes::LOGICAL_ERROR);
 
-    header = output_ports.front()->getHeader();
-    for (size_t i = 1; i < output_ports.size(); ++i)
-        assertBlocksHaveEqualStructure(header, output_ports[i]->getHeader(), "Pipe");
+    if (check_block_structure)
```

Comment:
I don't like this flag, cause it is brakes internal invariant - all output ports must have the same header.
If possible, can we avoid it?

Reply:
Well, this flag is necessary in case when invariant is 100% broken. For example, after adding `AggregatingTransform` for each set we have different headers in output ports.